### PR TITLE
Added a publisher for estimator status in sys_status.

### LINKED ISSUE #1348: No plugin publishing for px4 estimator status.
### Issue details

Hi, my work involves using the px4-mavros integration for uavs in simulations. For that I require to  repeatedly switch on/off the estimator after every simulation iteration for which I use the px4-ekf2/local_position_estimator binaries. However I found no reliable way to get the current status of the estimator and to check whether the estimator has stabilized and started giving outputs with valid states in mavros. I saw that mavlink already provides a message for that but its not being published by any mavros plugin.

To solve the problem I've made a estimator_status plugin that publishes it with the following message:

```std_msgs/Header header
bool stable_attitude
bool stable_hor_velocity_est
bool stable_ver_velocity_est
bool stable_hor_position_est_rel
bool stable_hor_position_est_abs
bool stable_ver_position_est_abs
bool stable_ver_position_est_agl
bool const_position_mode
bool pred_hor_position_rel
bool pred_hor_position_abs
bool gps_glitch
bool accel_error```

Each bool represents the corresponding ESTIMATOR_STATUS_FLAGS defined in mavlink message definitions.h. Is this the right way to solve the problem or is there any other way? If yes then can I PR this addition?

### MAVROS version and platform

Mavros: ?0.18.4?
ROS: ?Melodic?
Ubuntu: ?18.04?


### Autopilot type and version

[  ] ArduPilot
[x] PX4

Version: ?3.7.1?

### PR DESCRIPTION
**Pull request by saifullah3396**
@TSC21 I believe this way is correct? Let me know if any changes are to be made.

Fixes #1348.

### CODE REVIEW

File: mavros_msgs/msg/EstimatorStatus.msg [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+bool stable_hor_position_est_rel
+bool stable_hor_position_est_abs
+bool stable_ver_position_est_abs
+bool stable_ver_position_est_agl
+bool const_position_mode
+bool pred_hor_position_rel
+bool pred_hor_position_abs
+bool gps_glitch
+bool accel_error
+
```
Comment (vooon) [inline mavros_msgs/msg/EstimatorStatus.msg]: Please add comment-header like in other messages with links to mavlink docs.

File: mavros/src/plugins/sys_status.cpp [hunk 2]
```diff
@@ -908,6 +912,29 @@ class SystemStatusPlugin : public plugin::PluginBase
...
+		est_status_msg->stable_hor_position_est_abs = !!(status.flags & utils::enum_value(mavlink::common::ESTIMATOR_STATUS_FLAGS::POS_HORIZ_ABS));
+		est_status_msg->stable_ver_position_est_abs = !!(status.flags & utils::enum_value(mavlink::common::ESTIMATOR_STATUS_FLAGS::POS_VERT_ABS));
+		est_status_msg->stable_ver_position_est_agl = !!(status.flags & utils::enum_value(mavlink::common::ESTIMATOR_STATUS_FLAGS::POS_VERT_AGL));
+		est_status_msg->const_position_mode = !!(status.flags & utils::enum_value(mavlink::common::ESTIMATOR_STATUS_FLAGS::CONST_POS_MODE));
+		est_status_msg->pred_hor_position_rel = !!(status.flags & utils::enum_value(mavlink::common::ESTIMATOR_STATUS_FLAGS::PRED_POS_HORIZ_REL));
+		est_status_msg->pred_hor_position_abs = !!(status.flags & utils::enum_value(mavlink::common::ESTIMATOR_STATUS_FLAGS::PRED_POS_HORIZ_ABS));
+		est_status_msg->gps_glitch = !!(status.flags & utils::enum_value(mavlink::common::ESTIMATOR_STATUS_FLAGS::GPS_GLITCH));
+		est_status_msg->accel_error = !!(status.flags & utils::enum_value(mavlink::common::ESTIMATOR_STATUS_FLAGS::ACCEL_ERROR));
```
Comment (vooon) [inline mavros/src/plugins/sys_status.cpp]: I'd prefer to generate all that fields with cog.

**Comment by saifullah3396** [comment]
Updated. Please see updates. @vooon

File: mavros_msgs/msg/EstimatorStatus.msg [hunk 3]
```diff
@@ -0,0 +1,17 @@
+# Current autopilot estimator state
+#
+# https://mavlink.io/en/messages/common.html#ESTIMATOR_STATUS_FLAGS
+
+std_msgs/Header header
+bool attitude_status_flag
```
Comment (TSC21) [inline mavros_msgs/msg/EstimatorStatus.msg]: @saifullah3396 please split the fields by subject. Like
```
bool velocity_horiz_status_flag
bool velocity_vert_status_flag

bool pos_horiz_rel_status_flag
bool pos_horiz_abs_status_flag

bool pos_vert_abs_status_flag
bool pos_vert_agl_status_flag
```

File: mavros/src/plugins/sys_status.cpp [hunk 4]
```diff
@@ -919,19 +919,40 @@ class SystemStatusPlugin : public plugin::PluginBase
...
+		//     desc = e.description.split(' ', 1)[1] if e.description.startswith('0x') else e.description
+		//     esf = e.name
+		//
+		//     if esf.startswith(ename + '_'):
+		//         esf = esf[len(ename) + 1:]
+		//     if esf.startswith(ename_pfx2):
+		//         esf = esf[len(ename_pfx2):]
+		//     if esf[0].isdigit():
+		//         esf = 'SENSOR_' + esf
+		//     cog.outl("est_status_msg->%s_status_flag = !!(status.flags & enum_value(ESF::%s));" % (esf.lower(), esf))
```
Comment (vooon) [inline mavros/src/plugins/sys_status.cpp]: Oh, i should later generalise enum-name shortening code...

File: mavros_msgs/msg/EstimatorStatus.msg (hunk 3 above)
Comment (TSC21) [reply mavros_msgs/msg/EstimatorStatus.msg]: @kshahar can you address this so we can merge it? Thanks!

Comment (saifullah3396) [reply mavros_msgs/msg/EstimatorStatus.msg]: Sorry, been on vacations! I'll do this tomorrow if it's not fixed until then.

**Pull request merged**